src/bin/cut.rs

```diff
@@ -0,0 +1,525 @@
+     dash (`-'), and a second number and select the fields or columns from the first number to the second, inclusive.  Numbers or number
+     ranges may be preceded by a dash, which selects all fields or columns from 1 to the last number.  Numbers or number ranges may be fol-
+     lowed by a dash, which selects all fields or columns from the last number to the end of the line.  Numbers and number ranges may be
+     repeated, overlapping, and in any order.  If a field or column is specified multiple times, it will appear only once in the output.  It
+     is not an error to select fields or columns not present in the input line.
+
+     The options are as follows:
+
+     -b list
+             The list specifies byte positions.
+
+     -c list
+             The list specifies character positions.
+
+     -d delim
+             Use delim as the field delimiter character instead of the tab character.
+
+     -f list
+             The list specifies fields, separated in the input by the field delimiter character (see the -d option.)  Output fields are sepa-
+             rated by a single occurrence of the field delimiter character.
+
+     -s      Suppress lines with no field delimiter characters.  Unless specified, lines with no delimiters are passed through unmodified.
+
+EXIT STATUS
+     The cut utility exits 0 on success, and >0 if an error occurs.
+
+EXAMPLES
+     Extract users' login names and shells from the system passwd(5) file as ``name:shell'' pairs:
+
+           cut -d : -f 1,7 /etc/passwd
+
+     Show the names and login times of the currently logged in users:
+
+           who | cut -c 1-16,26-38
+
+AUTHOR
+     Written by Hernán E. Grecco.
+"#;
+
+/// The Selection object.
+#[derive(Debug, PartialEq)]
+struct Selection {
+    /// Indicates which elements are selected.
+    selected: Vec<bool>,
+    /// True indicates that the selection continues after the range covered by `selected`.
+    to_eol: bool,
+}
+
+impl Selection {
+    /// Constructs a Selection object from a list option argument.
+    ///
+    /// The list option argument is a comma or whitespace separated set of numbers and/or
+    /// number ranges.  Number ranges consist of a number, a dash (`-'), and a second number
+    /// and select the fields or columns from the first number to the second, inclusive.
+    /// Numbers or number ranges may be preceded by a dash, which selects all fields or columns
+    /// from 1 to the last number.  Numbers or number ranges may be followed by a dash,
+    /// which selects all fields or columns from the last number to the end of the line.
+    /// Numbers and number ranges may be repeated, overlapping, and in any order.
+    /// If a field or column is specified multiple times, it will appear only once in the output.
+    fn from_str(list: &str) -> Result<Self, ()> {
+
+        let empty = usize::from_str("");
+
+        let mut selected: Vec<bool> = vec![];
```

Comment:
1. You don't need to denote the type here. Type inference does the job.

2. vec![] is an anti-pattern, prefer `Vec::new()` instead, or ideally `Vec::with_capacity()`, since you got a decent estimate in this case.